controllers/users/users.js

```diff
@@ -0,0 +1,61 @@
+const { validationResult } = require("express-validator/check");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const config = require("config");
+const User = require("../../models/User");
+
+exports.registerUser = (req, res) => {
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    return res
+      .status(400)
+      .json({ errors: errors.array({ onlyFirstError: true }) });
+  }
+
+  const { firstName, lastName, email, password } = req.body;
+
+  User.findOne({ email: email.toLowerCase() }).then(user => {
+    if (user) {
+      return res
+        .status(400)
+        .json({ errors: [{ msg: "Email already exists" }] });
+    } else {
+      const newUser = new User({
```

Comment:
Why do we need to create another instance of user object ? Can't we pass just req.body ?

Reply:
I'm not sure if i am uderstanding what do you asking. We need(?) to pass data through new User model to save user in users collection,